File: projetos-fenix-para-estudo/t2ti-erp-fenix-cadastros-completo/fontes/frontend/mvc/lib/src/model/pessoa_fisica.dart
class PessoaFisica implements Comparable<PessoaFisica> {
  PessoaFisica();

  String 
        _id,
        _idPessoa,
        _idNivelFormacao,
        _idEstadoCivil,
        _cpf,
        _rg,
        _orgaoRg,
        _dataEmissaoRg,
        _dataNascimento,
        _sexo,
        _raca,
        _nacionalidade,
        _naturalidade,
        _nomePai,
        _nomeMae;

  String get id => _id;
  set id(String value){
    if (value == null) value = "";
    _id = value;
  }

  String get idPessoa => _idPessoa;
  set idPessoa(String value){
    if (value == null) value = "";
    _idPessoa = value;
  }

  String get idNivelFormacao => _idNivelFormacao;
  set idNivelFormacao(String value){
    if (value == null) value = "";
    _idNivelFormacao = value;
  }

  String get idEstadoCivil => _idEstadoCivil;
  set idEstadoCivil(String value){
    if (value == null) value = "";
    _idEstadoCivil = value;
  }

  String get cpf => _cpf;
  set cpf(String value){
    if (value == null) value = "";
    _cpf = value;
  }

  String get rg => _rg;
  set rg(String value){
    if (value == null) value = "";
    _rg = value;
  }

  String get orgaoRg => _orgaoRg;
  set orgaoRg(String value){
    if (value == null) value = "";
    _orgaoRg = value;
  }

  String get dataEmissaoRg => _dataEmissaoRg;
  set dataEmissaoRg(String value){
    if (value == null) value = "";
    _dataEmissaoRg = value;
  }

  String get dataNascimento => _dataNascimento;
  set dataNascimento(String value){
    if (value == null) value = "";
    _dataNascimento = value;
  }

  String get sexo => _sexo;
  set sexo(String value){
    if (value == null) value = "";
    _sexo = value;
  }

  String get raca => _raca;
  set raca(String value){
    if (value == null) value = "";
    _raca = value;
  }

  String get nacionalidade => _nacionalidade;
  set nacionalidade(String value){
    if (value == null) value = "";
    _nacionalidade = value;
  }

  String get naturalidade => _naturalidade;
  set naturalidade(String value){
    if (value == null) value = "";
    _naturalidade = value;
  }

  String get nomePai => _nomePai;
  set nomePai(String value){
    if (value == null) value = "";
    _nomePai = value;
  }

  String get nomeMae => _nomeMae;
  set nomeMae(String value){
    if (value == null) value = "";
    _nomeMae = value;
  }

  PessoaFisica.fromMap(Map m) {
    id = m["id"];
    idNivelFormacao = m["idNivelFormacao"];
    idEstadoCivil = m["idEstadoCivil"];
    cpf = m["cpf"];
    rg = m["rg"];
    orgaoRg = m["orgaoRg"];
    dataEmissaoRg = m["dataEmissaoRg"];
    dataNascimento = m["dataNascimento"];
    sexo = m["sexo"];
    raca = m["raca"];
    nacionalidade = m["nacionalidade"];
    naturalidade = m["naturalidade"];
    nomePai = m["nomePai"];
    nomeMae = m["nomeMae"];
  }

  Map<String, dynamic> get toMap {
    return {
      "id":_id,
      "idNivelFormacao":_idNivelFormacao,
      "idEstadoCivil":_idEstadoCivil,
      "cpf":_cpf,
      "rg":_rg,
      "orgaoRg":_orgaoRg,
      "dataEmissaoRg":_dataEmissaoRg,
      "dataNascimento":_dataNascimento,
      "sexo":_sexo,
      "raca":_raca,
      "nacionalidade":_nacionalidade,
      "naturalidade":_naturalidade,
      "nomePai":_nomePai,
      "nomeMae":_nomeMae,
    };
  }  

  @override
  int compareTo(PessoaFisica other) => _idEstadoCivil.compareTo(other._idEstadoCivil);
}

File: projetos-fenix-para-estudo/t2ti-erp-fenix-cadastros-completo/fontes/backend/java/fenix/src/main/java/com/t2ti/fenix/model/cadastros/PessoaFisica.java
/*******************************************************************************
Title: T2Ti ERP Fenix                                                                
Description: Model relacionado à tabela [PESSOA_FISICA] 
                                                                                
The MIT License                                                                 
                                                                                
Copyright: Copyright (C) 2020 T2Ti.COM                                          
                                                                                
Permission is hereby granted, free of charge, to any person                     
obtaining a copy of this software and associated documentation                  
files (the "Software"), to deal in the Software without                         
restriction, including without limitation the rights to use,                    
copy, modify, merge, publish, distribute, sublicense, and/or sell               
copies of the Software, and to permit persons to whom the                       
Software is furnished to do so, subject to the following                        
conditions:                                                                     
                                                                                
The above copyright notice and this permission notice shall be                  
included in all copies or substantial portions of the Software.                 
                                                                                
THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,                 
EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES                 
OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND                        
NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT                     
HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,                    
WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING                    
FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR                   
OTHER DEALINGS IN THE SOFTWARE.                                                 
                                                                                
       The author may be contacted at:                                          
           t2ti.com@gmail.com                                                   
                                                                                
@author Albert Eije (alberteije@gmail.com)                    
@version 1.0.0
*******************************************************************************/
package com.t2ti.fenix.model.cadastros;

import java.io.Serializable;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.NamedQuery;
import javax.persistence.Table;
import java.util.Date;
import javax.persistence.Temporal;
import javax.persistence.TemporalType;
import javax.persistence.JoinColumn;
import javax.persistence.OneToOne;
import javax.persistence.ManyToOne;
import com.fasterxml.jackson.annotation.JsonIgnore;

@Entity
@Table(name="PESSOA_FISICA")
@NamedQuery(name="PessoaFisica.findAll", query="SELECT t FROM PessoaFisica t")
public class PessoaFisica implements Serializable {
	private static final long serialVersionUID = 1L;

    @Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Integer id;

    @Column(name="CPF")
	private String cpf;

    @Column(name="RG")
	private String rg;

    @Column(name="ORGAO_RG")
	private String orgaoRg;

    @Temporal(TemporalType.DATE)
	@Column(name="DATA_EMISSAO_RG")
	private Date dataEmissaoRg;

    @Temporal(TemporalType.DATE)
	@Column(name="DATA_NASCIMENTO")
	private Date dataNascimento;

    @Column(name="SEXO")
	private String sexo;

    @Column(name="RACA")
	private String raca;

    @Column(name="NACIONALIDADE")
	private String nacionalidade;

    @Column(name="NATURALIDADE")
	private String naturalidade;

    @Column(name="NOME_PAI")
	private String nomePai;

    @Column(name="NOME_MAE")
	private String nomeMae;

    @OneToOne
	@JsonIgnore
	@JoinColumn(name="ID_PESSOA")
    private Pessoa pessoa;

    @ManyToOne
	@JoinColumn(name="ID_NIVEL_FORMACAO")
    private NivelFormacao nivelFormacao;

    @ManyToOne
	@JoinColumn(name="ID_ESTADO_CIVIL")
    private EstadoCivil estadoCivil;

	public PessoaFisica() {
	}

    public Integer getId() {
		return this.id;
	}

	public void setId(Integer id) {
		this.id = id;
	}

    public String getCpf() {
		return this.cpf;
	}

	public void setCpf(String cpf) {
		this.cpf = cpf;
	}

    public String getRg() {
		return this.rg;
	}

	public void setRg(String rg) {
		this.rg = rg;
	}

    public String getOrgaoRg() {
		return this.orgaoRg;
	}

	public void setOrgaoRg(String orgaoRg) {
		this.orgaoRg = orgaoRg;
	}

    public Date getDataEmissaoRg() {
		return this.dataEmissaoRg;
	}

	public void setDataEmissaoRg(Date dataEmissaoRg) {
		this.dataEmissaoRg = dataEmissaoRg;
	}

    public Date getDataNascimento() {
		return this.dataNascimento;
	}

	public void setDataNascimento(Date dataNascimento) {
		this.dataNascimento = dataNascimento;
	}

    public String getSexo() {
		return this.sexo;
	}

	public void setSexo(String sexo) {
		this.sexo = sexo;
	}

    public String getRaca() {
		return this.raca;
	}

	public void setRaca(String raca) {
		this.raca = raca;
	}

    public String getNacionalidade() {
		return this.nacionalidade;
	}

	public void setNacionalidade(String nacionalidade) {
		this.nacionalidade = nacionalidade;
	}

    public String getNaturalidade() {
		return this.naturalidade;
	}

	public void setNaturalidade(String naturalidade) {
		this.naturalidade = naturalidade;
	}

    public String getNomePai() {
		return this.nomePai;
	}

	public void setNomePai(String nomePai) {
		this.nomePai = nomePai;
	}

    public String getNomeMae() {
		return this.nomeMae;
	}

	public void setNomeMae(String nomeMae) {
		this.nomeMae = nomeMae;
	}

    public Pessoa getPessoa() {
		return this.pessoa;
	}

	public void setPessoa(Pessoa pessoa) {
		this.pessoa = pessoa;
	}

    public NivelFormacao getNivelFormacao() {
		return this.nivelFormacao;
	}

	public void setNivelFormacao(NivelFormacao nivelFormacao) {
		this.nivelFormacao = nivelFormacao;
	}

    public EstadoCivil getEstadoCivil() {
		return this.estadoCivil;
	}

	public void setEstadoCivil(EstadoCivil estadoCivil) {
		this.estadoCivil = estadoCivil;
	}

		
}